modules/api/src/main/java/org/eclipse/xpanse/api/controllers/IsvServiceDeployApi.java

```diff
@@ -97,6 +98,22 @@ public ServiceDetailVo getServiceDetailsByIdForIsv(
         return this.deployService.getServiceDetailsByIdForIsv(UUID.fromString(id));
     }
 
+    /**
+     * Get details of the managed VendorHosted service by id.
+     *
+     * @return VendorHostedServiceDetailsVo of the managed service.
+     */
+    @Tag(name = "Service", description = "APIs to manage the service instances")
+    @Operation(description = "Get deployed service details by id.")
+    @GetMapping(value = "/isv/service/vendor_hosted/{id}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public VendorHostedServiceDetailsVo getVendorHostedServiceDetailsById(
+            @Parameter(name = "id", description = "Task id of deployed service")
+            @PathVariable("id") String id) {
+        return this.deployService.getVendorHostedServiceDetailsById(UUID.fromString(id));
```

Comment:
@WangLiNaruto this is a service for the end user to get details of vendor hosted service. So this must be allowed for role "user" and must be part of that controller class.